### PR TITLE
Overhead tests setup

### PR DESCRIPTION
**Pull request by lachmatt**
## Why

Run automated test to measure overhead of agent in different configurations.

## What

This setup follows the way outlined in [java tests](https://github.com/signalfx/splunk-otel-java-overhead-test) related to provisioning/test organization, with minimal modifications.

- provisioning scripts adapted from [java tests](https://github.com/signalfx/splunk-otel-java-overhead-test) 
- automated test that uses `testcontainers` to create and cleanup required infrastucture 

## Tests

Included in PR.

### CODE REVIEW

File: overhead-test/provision/README.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+The overhead tests attempt to reduce the impact of noisy neighbors by provisioning
+the collector and sqlserver services on a separate cloud instance.
+
+This directory contains automation tooling used to provision and configure 
+the EC2 instances used for this test. It uses an internal Splunk tool 
+called [Orca](https://core-ee.splunkdev.page/orca/) to do the provisioning 
+and ansible to automatically configure the VMs.
+
+# setup
```
Comment (Kielek) [inline overhead-test/provision/README.md:9]:
```suggestion
# Setup
```

and similar for following headers

File: overhead-test/provision/ansible/externals-playbook.yml [hunk 2]
```diff
@@ -0,0 +1,53 @@
+---
+- hosts: externals
+  become: true
+  tasks:
+    - name: install iptables
+      apt:
+        name:
+          - iptables
+    - name: lock down the docker port so only the testbox can talk to it
```
Comment (Kielek) [inline overhead-test/provision/ansible/externals-playbook.yml:9]: Names in other yml files start with capital letter

**Review by Kielek — APPROVED** [review]

**Review by Kielek — APPROVED** [review]

File: overhead-test/src/SignalFx.OverheadTest.csproj [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+  <ItemGroup>
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="16.11.0"/>
+    <PackageReference Include="xunit" Version="2.4.1"/>
+    <PackageReference Include="DotNet.Testcontainers" Version="1.5.0"/>
+    <PackageReference Include="xunit.runner.visualstudio" Version="2.4.3">
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
+    <PackageReference Include="coverlet.collector" Version="3.1.0">
```
Comment (pjanotti) [inline overhead-test/src/SignalFx.OverheadTest.csproj:17]: Q.: are we collecting code coverage for this project?

File: overhead-test/src/SignalFx.OverheadTest.csproj [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+      <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
+      <PrivateAssets>all</PrivateAssets>
+    </PackageReference>
+    <PackageReference Update="Microsoft.NET.Test.Sdk" Version="17.2.0"/>
```
Comment (pjanotti) [inline overhead-test/src/SignalFx.OverheadTest.csproj:21]: Multiple versions of this and the package below.

**Review by pjanotti — APPROVED** [review]
LGTM @lachmat minus a couple of minor things. It would be good to build the new csproj during CI in a follow up PR.

File: overhead-test/src/SignalFx.OverheadTest.csproj (hunk 3 above)
Comment (lachmatt) [reply overhead-test/src/SignalFx.OverheadTest.csproj:17]: We are not, leftover after creating project from default template. Fixed in https://github.com/signalfx/signalfx-dotnet-tracing/pull/510

File: overhead-test/src/SignalFx.OverheadTest.csproj (hunk 4 above)
Comment (lachmatt) [reply overhead-test/src/SignalFx.OverheadTest.csproj:21]: fixed in https://github.com/signalfx/signalfx-dotnet-tracing/pull/510

File: overhead-test/provision/README.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+* Follow the docs and links here [to request access](https://core-ee.splunkdev.page/orca/docs/providers/aws#through-cli).
+
+## Install orca
+
+Orca is a Splunk-internal tool for provisioning cloud instances.
+[Go here to learn how to set it up](https://core-ee.splunkdev.page/orca/docs/setup).
+
+## Install ansible
+
+Use e.g `pip` to install [ansible](https://docs.ansible.com/ansible/latest/installation_guide/intro_installation.html#)
```
[the remark this answers is not in the document]
Comment (lachmatt) [reply overhead-test/provision/README.md:21]: fixed in https://github.com/signalfx/signalfx-dotnet-tracing/pull/510

**Pull request merged**